File: C#/atividades/atividade4/FormasGeometricas/FormasGeometricas/Models/Quadrado.cs
namespace FormasGeometricas.FormaGenerica;

internal class Quadrado : FormaGenerica
{
    public float Altura { get; set; }
    public float Base { get; set; }

    public Quadrado(string nome, float altura, float base1) : base(nome)
    {
        Altura = altura;
        Base = base1;
    }

    public override void ExibirInfo()
    {
        base.ExibirInfo();
        CalculoArea();
    }

    public void CalculoArea()
    {
       Console.WriteLine($"Área do {Nome}: {Altura * Base}");
    }
}

File: JAVA/Atividades/orientacao-a-objeto/video2/atividade  4 ok/FormasGeometricas/src/Quadrado.java
public class Quadrado extends FormaGeometrica implements ICalculavel {
    private double Altura;
    private double Base;

    public Quadrado(String cor, String nome, double altura, double base) {
        super(cor, nome);

        this.Altura = altura;
        this.Base = base;
    }

    public double getAltura() {
        return Altura;
    }

    public void setAltura(double altura) {
        Altura = altura;
    }

    public double getBase() {
        return Base;
    }

    public void setBase(double base) {
        Base = base;
    }


    @Override
    public void calcularArea() {
        System.out.println("Área do quadrado: " + this.Altura * this.Altura);
    }

    @Override
    public void calcularPerimetro() {
        System.out.println("Perímetro do quadrado: " + this.Altura * 4);
    }
}
